easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1226,6 +1226,29 @@ def get_parsed_multi_deps(self):
 
         return multi_deps
 
+    # new function/method
+    def find_dep_version_match(self, dep_version_candidates):
+        """Identify the correct version for this system from the choices provided"""
+        # figure out matches based on dict keys (after splitting on '=')
+        matches = []
+        my_arch = get_cpu_architecture()
+        for key, value in dep_version_candidates.items():
+            if '=' in key:
+                inner_key, inner_value = key.split('=', 1)
+                if inner_key == 'arch' and inner_value == my_arch:
```

Comment:
This could be done more simply via
```
if key == 'arch='+my_arch:
```
You could also iterate `key in dep_version_candidates:`, and then use `dep_version_candidate[key]` for `value`. That is more efficient in Python 2 mainly (iterator vs copy)

Reply:
or even
```
matches = [dep_version_candidates[key] for key in dep_version_candidates if key == 'arch'+my_arch]
```

Reply:
I've added a test - then if that passes on here I'll look at the changes.

Reply:
Simplified the code further, having thought for 1 minute about what a dictionary is...